src/main/java/de/danielbechler/diff/ObjectDifferBuilder.java

```diff
@@ -160,7 +160,8 @@ public ObjectDiffer build()
 				circularReferenceService,
 				inclusionService,
 				returnableNodeService,
-				introspectionService);
+				introspectionService,
+				inclusionService);
```

Comment:
You can just inject the `categoryService` here. That way the InclusionService doesn't need to implement the `CategoryResolver` interface.